libs/barista-components/autocomplete/src/autocomplete-trigger.ts

```diff
@@ -297,6 +314,22 @@ export class DtAutocompleteTrigger<T>
           }
         });
     }
+
+    if (_optionConfig) {
+      const height =
+        _optionConfig.height > AUTOCOMPLETE_OPTION_HEIGHT
+          ? _optionConfig.height
+          : AUTOCOMPLETE_OPTION_HEIGHT;
+
+      this._optionHeight = height;
+      this._panelMaxHeight =
+        AUTOCOMPLETE_PANEL_MAX_HEIGHT % height === 0
+          ? AUTOCOMPLETE_PANEL_MAX_HEIGHT - height / 2
+          : AUTOCOMPLETE_PANEL_MAX_HEIGHT;
```

Comment:
Write some unit tests for it to test the calculation.

NIT: it would be nice and easier to test to put it in a pure function.